src/logger.ts

```diff
@@ -12,36 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import * as consoleLogLevel from 'console-log-level';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
 import {defaultConfig} from './config';
+import {LogSync, Logging} from '@google-cloud/logging-min';
+
+const logging = new Logging();
+
+logging.setProjectId().catch(err => {
+  console.error(`failed to set logging project id ${err}`);
+});
+logging.setDetectedResource().catch(err => {
+  console.error(`failed to discover resource metadata ${err}`);
+});
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const pjson = require('../../package.json');
 
-const LEVEL_NAMES: consoleLogLevel.LogLevelNames[] = [
-  'fatal',
-  'error',
-  'warn',
-  'info',
-  'debug',
-  'trace',
-];
-
-function logLevelToName(level?: number): consoleLogLevel.LogLevelNames {
-  if (level === undefined) {
-    level = defaultConfig.logLevel;
-  } else if (level < 0) {
-    level = 0;
-  } else if (level > 4) {
-    level = 4;
+export class Logger {
+  private log: LogSync;
+  private severityThreshold: number;
+
+  constructor(readonly level?: number) {
+    if (level === undefined) {
+      level = defaultConfig.logLevel;
+    }
+    if (level < 0) {
```

Comment:
I guess you should define a constants for this

Reply:
Can you be more specific? What values should be set to constants?

Reply:
Added constants and a comment that explains where it comes from.